### DISCUSSION TITLE
Custom field in script

### DISCUSSION THREAD

**Discussion by vaas2956** [netbox-community/netbox#13218] [Q&A]
Hello team,

For a creation of new site, I want to import the data of another site (rack, devices ...) in this new site.
To do this, I created this first script (to add rack) : 

---------------------------------------------------------------------------------

from django.utils.text import slugify
from extras.scripts import *
from dcim.choices import DeviceStatusChoices, SiteStatusChoices, RackStatusChoices, LocationStatusChoices
from dcim.models import Device, DeviceRole, DeviceType, Manufacturer, Site, Rack, Location, Interface
from extras.scripts import *

class test(Script):

    class Meta:
        name = "Add rack"
        description = "(update)"

    site_name = ObjectVar(
        description="Site to update",
        model=Site,
        required=True
    )

    def run(self, data, commit):

        ### VARIABLES ###
        siteToUpdate = Site.objects.get(name=data['site_name'])
        siteOld = Site.objects.get(name="old")
        allRack = []
        
        ### INIT ###
        for rack in Rack.objects.filter(site=siteOld):
            allRack.append(rack)

        for i in allRack:
            rackRet = Rack(
                site = siteToUpdate,
                location = i.location,
                tenant = i.tenant,
                # niv = i.custom_field_data.get('niv'),
                # rtpf = i.custom_field_data.get('rtpf')

            )

            rackRet.full_clean()
            rackRet.save()
            self.log_success(f"Created new rack: {i}")

------------------------------------------------------------------

However I use customs fields to initialize the rack and I don't know how initialize this in rack's constructor ?
Do you know please how I can use customs fields in the constructor ? 
Thanks

Regards

**Comment by PaulR282** [comment]
To fill custom data in the `rackSet` you need to: 
```
rackSet.custom_field_data = {
      "cf_name": "string_to_be_shown",
}
